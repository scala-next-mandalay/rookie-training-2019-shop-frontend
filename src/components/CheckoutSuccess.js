import React from 'react';
import { makeStyles } from '@material-ui/core/styles'
import {  Container, Paper,Box,Button} from '@material-ui/core';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import TitleBar from '../containers/TitleBar';
import PropTypes from 'prop-types';
import '../assets/style.css';
import { withRouter } from 'react-router-dom';
import HomeIcon from '@material-ui/icons/Home';
import { FormattedMessage } from 'react-intl';


const useStyles = makeStyles(theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing(3),
    overflowX: 'auto',
  },
  table: {
    minWidth: 650,
  },
  tbHead: {
    backgroundColor:'#f2f2f2',
  },
  toolbar:{
   marginTop: theme.spacing(10),
  }
}));


const CheckoutConfirm = ({postResultObj,history}) => {
  const classes = useStyles();
  
  const handleHome = event => {
    event.preventDefault();
    history.push("/");
  };
  
  return (
    <Container maxWidth="lg">
      <TitleBar showMenu={false} showIcon={false} showNav={false}/>
      <div className={classes.toolbar}/>
      
      <Box display="flex" >
        <Box ml="auto" my="auto" mr={1}>
          <Button onClick={handleHome} variant="contained" color="primary">
            <HomeIcon className={classes.homeSpace} />
            <FormattedMessage id="Button.Shopping" defualtMessage="Shopping" />
          </Button>
        </Box>
      </Box>
         
      <Paper className={classes.root}>
      
        <div className="title_Ship">
          <div className="ho"><FormattedMessage id="Label.OrderSuccess"/></div>
          <div className="text2"><FormattedMessage id="Label.OrderSuccessId"/> {postResultObj.id}</div>
          <div className="text1"><FormattedMessage id="Label.InfoDetail"/></div>
        </div>
        <Table className={classes.table}>
          <TableHead className={classes.tbHead}>
            <TableRow>
               <TableCell align="left" >FirstName</TableCell>
               <TableCell align="left" >LastName</TableCell>
               <TableCell align="left" >Address1</TableCell>
               <TableCell align="left" >Address2</TableCell>
               <TableCell align="left" >Country</TableCell>
               <TableCell align="left" >State</TableCell>
               <TableCell align="left" >City</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
              <TableRow >
               <TableCell align="left"><div className="text2">{postResultObj!==null?postResultObj.first_name:null}</div></TableCell>
               <TableCell align="left"><div className="text2">{postResultObj!==null?postResultObj.last_name:null}</div></TableCell>
               <TableCell align="left"><div className="text2">{postResultObj!==null?postResultObj.address1:null}</div></TableCell>
               <TableCell align="left"><div className="text2">{postResultObj!==null?postResultObj.address2:null}</div></TableCell>
               <TableCell align="left"><div className="text2">{postResultObj!==null?postResultObj.country:null}</div></TableCell>
               <TableCell align="left"><div className="text2">{postResultObj!==null?postResultObj.state:null}</div></TableCell>
               <TableCell align="left"><div className="text2">{postResultObj!==null?postResultObj.city:null}</div></TableCell>
              </TableRow>
            <TableRow>
              <TableCell rowSpan={6} />
              <TableCell colSpan={5}><div className='text1'><FormattedMessage id="Label.SubTotal"/></div></TableCell>
              <TableCell align="left"><div className='text1'>{postResultObj!==null?postResultObj.total_price:null}MMK</div></TableCell>
            </TableRow>
            <TableRow>
              <TableCell colSpan={5}><div className='text1'><FormattedMessage id="Label.Tax"/></div></TableCell>
              <TableCell align="left"><Box className='text1'>0MMK</Box></TableCell>
            </TableRow>
            <TableRow>
              <TableCell colSpan={5}><div className='text1'><FormattedMessage id="Label.TotalPrice"/></div></TableCell>
              <TableCell align="left"><div className='text1'>{postResultObj!==null?postResultObj.total_price:null}MMK</div></TableCell>
            </TableRow>
          </TableBody>

        </Table>
      </Paper>
    </Container>
  );
};

CheckoutConfirm.propTypes = {
   postResultObj: PropTypes.object.isRequired,
};
export default withRouter(CheckoutConfirm);

