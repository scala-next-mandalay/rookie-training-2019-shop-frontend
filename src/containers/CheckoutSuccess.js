import { connect } from 'react-redux';
import CheckoutSuccess from '../components/CheckoutSuccess';
export default connect(
  (state) => ({
    postResultObj: state.orders.postResultObj
  })
)(CheckoutSuccess);
