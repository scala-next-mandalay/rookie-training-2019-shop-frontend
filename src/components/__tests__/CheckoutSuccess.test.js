import React from 'react'
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import CheckoutSuccess from '../CheckoutSuccess';
import Parent from '../Parent';
/*global expect*/
/*global jest*/
/*global Event*/
  let container;
  beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
  });


  afterEach(() => {
  document.body.outerHTML = '<body></body>';
  container = null;
  });
  const postResultObj = {
  id:"5",
  first_name: "khin",
  last_name: "lawoon",
  address1: "23 street",
  address2: "CMTZ",
  country: "myanmar",
  state: "mandalay",
  city: "mandalay",
  total_price: 50000
  }
   const history = { push: jest.fn()};
  
    
  // });
  describe("CheckoutSuccess  Component", () => {
  it('matches the snapshot', () => {
    const CheckoutSuccessSnapshotWithPostObj = renderer.create(
      
        <Parent>
          <CheckoutSuccess
            postResultObj={postResultObj}
            history={history}
          />
        </Parent>
    ).toJSON();
    expect(CheckoutSuccessSnapshotWithPostObj).toMatchSnapshot();
  });
 
  });

  describe("Testing Checkout Confirm  Component", () => {
  it('testing checkout order confirm page', () => {
    act(() => {
      ReactDOM.render((
        
      <Parent>
        <CheckoutSuccess
            postResultObj={postResultObj}
            history={history}
          />
      </Parent>
  
      ), container);
    });
    
    //testing order id 
    const divArr = container.querySelectorAll('div .text2');
    expect(divArr[0].textContent).toBe("Your Order Id is 5"); 
     
    /*start table*/
    /*start testing table head*/
    const thArr = container.querySelectorAll('table th');
    expect(thArr[0].textContent.toUpperCase()).toBe('FIRSTNAME');
    expect(thArr[1].textContent.toUpperCase()).toBe('LASTNAME');
    expect(thArr[2].textContent.toUpperCase()).toBe('ADDRESS1');
    expect(thArr[3].textContent.toUpperCase()).toBe('ADDRESS2');
    expect(thArr[4].textContent.toUpperCase()).toBe('COUNTRY');
    expect(thArr[5].textContent.toUpperCase()).toBe('STATE');
    expect(thArr[6].textContent.toUpperCase()).toBe('CITY');
    /*end table head*/
    
    /*start testing table body*/
    const tdArr = document.querySelectorAll('table tbody td');
    expect(tdArr[0].textContent).toBe("khin");
    expect(tdArr[1].textContent).toBe("lawoon");
    expect(tdArr[2].textContent).toBe("23 street");
    expect(tdArr[3].textContent).toBe("CMTZ");
    expect(tdArr[4].textContent).toBe("myanmar");
    expect(tdArr[5].textContent).toBe("mandalay");
    expect(tdArr[6].textContent).toBe("mandalay");
    expect(tdArr[8].textContent.toUpperCase()).toBe('SUBTOTAL');
    expect(tdArr[9].textContent).toBe("50000MMK");
    expect(tdArr[10].textContent.toUpperCase()).toBe('SALES TAX');
    expect(tdArr[11].textContent).toBe("0MMK");
    expect(tdArr[12].textContent.toUpperCase()).toBe('TOTAL PRICE');
    expect(tdArr[13].textContent).toBe("50000MMK");
    /*end table body*/
    
    const buttonArr = document.querySelectorAll('button');
    // console.log("buttonArr",buttonArr.length)
    // console.log("buttonArr",buttonArr[0].textContent)
    // console.log("buttonArr",buttonArr[1].textContent)
    // console.log("buttonArr",buttonArr[2].textContent)
    act(() => {
        buttonArr[2].dispatchEvent(new Event('click', { bubbles: true }))
      });
    expect(buttonArr[2].textContent.toUpperCase()).toBe("SHOPPING")

});
    
});