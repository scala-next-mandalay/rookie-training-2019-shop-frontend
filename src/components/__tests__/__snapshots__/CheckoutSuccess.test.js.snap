// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutSuccess  Component matches the snapshot 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div>
    <header
      className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-12 MuiAppBar-colorPrimary mui-fixed"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="Open drawer"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-13 MuiIconButton-edgeStart"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          />
        </button>
        <div
          className="PrivateHiddenCss-smDown-78"
        >
          <div
            className="MuiBox-root MuiBox-root-89 title"
          >
            <span>
              Rookie Training 2019
            </span>
          </div>
        </div>
        <div
          className="MuiBox-root MuiBox-root-90"
        >
          <div
            className="MuiGrid-root MuiGrid-container"
          >
            <div
              className="MuiBox-root MuiBox-root-194 makeStyles-navMenuList-17"
            />
            <button
              aria-controls="long-menu"
              aria-haspopup="true"
              aria-label="More"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    className="makeStyles-toolbar-4"
  />
  <div
    className="MuiBox-root MuiBox-root-219"
  >
    <div
      className="MuiBox-root MuiBox-root-220"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
          <span>
            Shopping
          </span>
        </span>
      </button>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation1 makeStyles-root-1 MuiPaper-rounded"
  >
    <div
      className="title_Ship"
    >
      <div
        className="ho"
      >
        <span>
          Your Order is Complete...Success...!
        </span>
      </div>
      <div
        className="text2"
      >
        <span>
          Your Order Id is
        </span>
         
        5
      </div>
      <div
        className="text1"
      >
        <span>
          Detail  Information
        </span>
      </div>
    </div>
    <table
      className="MuiTable-root makeStyles-table-2"
    >
      <thead
        className="MuiTableHead-root makeStyles-tbHead-3"
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
            scope="col"
          >
            FirstName
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
            scope="col"
          >
            LastName
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
            scope="col"
          >
            Address1
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
            scope="col"
          >
            Address2
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
            scope="col"
          >
            Country
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
            scope="col"
          >
            State
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
            scope="col"
          >
            City
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
      >
        <tr
          className="MuiTableRow-root"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
          >
            <div
              className="text2"
            >
              khin
            </div>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
          >
            <div
              className="text2"
            >
              lawoon
            </div>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
          >
            <div
              className="text2"
            >
              23 street
            </div>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
          >
            <div
              className="text2"
            >
              CMTZ
            </div>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
          >
            <div
              className="text2"
            >
              myanmar
            </div>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
          >
            <div
              className="text2"
            >
              mandalay
            </div>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
          >
            <div
              className="text2"
            >
              mandalay
            </div>
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
            rowSpan={6}
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
            colSpan={5}
          >
            <div
              className="text1"
            >
              <span>
                SubTotal
              </span>
            </div>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
          >
            <div
              className="text1"
            >
              50000
              MMK
            </div>
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
            colSpan={5}
          >
            <div
              className="text1"
            >
              <span>
                Sales Tax
              </span>
            </div>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
          >
            <div
              className="MuiBox-root MuiBox-root-257 text1"
            >
              0MMK
            </div>
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
            colSpan={5}
          >
            <div
              className="text1"
            >
              <span>
                Total Price
              </span>
            </div>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
          >
            <div
              className="text1"
            >
              50000
              MMK
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
